crates/turborepo-lib/src/run/mod.rs

```diff
@@ -289,16 +290,32 @@ impl Run {
             )
         };
 
-        let mut pkg_dep_graph =
-            PackageGraph::builder(&self.base.repo_root, root_package_json.clone())
-                .with_single_package_mode(self.opts.run_opts.single_package)
-                .with_package_discovery(FallbackPackageDiscovery::new(
-                    daemon.as_mut().map(DaemonPackageDiscovery::new),
-                    fallback,
-                    duration,
-                ))
-                .build()
-                .await?;
+        // Unused until we set the daemon to be the default
+        let _fallback = FallbackPackageDiscovery::new(
+            daemon.as_mut().map(DaemonPackageDiscovery::new),
+            fallback,
+            duration,
+        );
+
+        let mut pkg_dep_graph = {
+            let builder = PackageGraph::builder(&self.base.repo_root, root_package_json.clone())
+                .with_single_package_mode(self.opts.run_opts.single_package);
+
+            #[cfg(feature = "daemon-package-discovery")]
+            let builder = builder.with_package_discovery(fallback);
+
+            #[cfg(not(feature = "daemon-package-discovery"))]
+            let builder = builder.with_package_discovery(
+                LocalPackageDiscoveryBuilder::new(
+                    self.base.repo_root.clone(),
+                    None,
+                    Some(root_package_json.clone()),
+                )
+                .build()?,
+            );
```

Comment:
I think this is unnecessary as this is what package discovery gets chosen by default if one isn't provided for `PackageGraphBuilder`
```suggestion
```